src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Receivers.kt

```diff
@@ -146,14 +142,14 @@ open class TaskReceiver(tasksDirectory: String,
 class MarkdownTaskReceiver(
         private val texCompiler: CompilerApi,
         tasksDirectory: String,
-        resultPublisher: PublisherApi,
-        private val config: Config = DEFAULT_CONFIG) : TaskReceiver(tasksDirectory, resultPublisher) {
+        private val resultPublisher: PublisherApi,
+        private val config: Config = DEFAULT_CONFIG) : CompilerFleetMessageReceiver() {
 
+    private val tasksDir: Path = getTaskDir(tasksDirectory)
     private val arguments = listOf("projectRootAbsPath", "workingDirRelPath",
             "inputFileName", "outputFileName", "mainFont")
     private val COMPILE_COMMAND_KEY = "pandoc.compile.command"
-    // TODO: fixedThreadPool? if so, how many threads?
-    private val executor = Executors.newFixedThreadPool(4)
+    private val executor = Executors.newSingleThreadExecutor()
```

Comment:
If you pass executor to the constructor, you will make testing easier (e.g. you will be able to pass [direct executor](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/util/concurrent/MoreExecutors.html#directExecutor()))

Reply:
Done